falcon/inspect.py

```diff
@@ -226,13 +239,13 @@ def _traverse(roots, parent):
 
                         source_info = _get_source_info(real_func)
                         internal = _is_internal(real_func)
-
+                        assert source_info
```

Comment:
Are these `assert`s essential to the inspect module's functionality, or debugging remnants?

Reply:
"_get_source_info" can return None or str, but in this case it's always a string. In such cases I just use asserts instead of handling never occuring cases.

Reply:
We could use `cast` or split _get_source_info to two functions with and without default. With default would always return str.

Reply:
That works for me, I was just wondering. Well, ideally you could maybe add `assert`ion messages explaining the error?

Reply:
Added message.